minicluster/src/main/java/alluxio/master/AbstractLocalAlluxioCluster.java

```diff
@@ -460,6 +466,7 @@ protected void setHostname() {
    * @throws IOException when the operation fails
    */
   protected void setAlluxioHome() throws IOException {
+    Preconditions.checkState(mHome == null);
```

Comment:
I see, how about a fancier shutdown hook which does the recursive delete? I think using a shutdown hook would be cleaner than relying on stop, but both ways do not give 100% guarantees.